examples/azure-rules.yaml

```diff
@@ -656,4 +656,16 @@ groups:
           Please, check the naming convention in case the resource configuration is done correctly.
            
           More information at https://docs.microsoft.com/en-us/azure/virtual-machines/workloads/sap/sap-hana-high-availability#create-sap-hana-cluster-resources
+        scored: true
+  - id: 1.6
+    description: "Check OS and package versions are supported and up-to-date"
+    checks:
+      - id: 1.6.1
+        description: "Check the base product"
+        audit: 'ls -la /etc/products.d/baseproduct | sed "s/^.*-> //"'
+        tests:
+          test_items:
+            - flag: 'SLES_SAP.prod'
```

Comment:
@aleksei-burlakov  As I have commented before, SLES is also a valid option for SAP Applications, so we should also consider vanilla SLES as a valid OS. 

On the remediation text, we need something more meaningful, but we can include this later, with the support from PM.

Reply:
@lee-martin I think we should also link on the remediation text the SAP Note of supported OS.

Reply:
As they say in https://www.suse.com/support/kb/doc/?id=000019341

_Even when the base product is SLES for SAP, the contents of the os-release file should never have been used to identify this fact._

I understand we should only look in the `/etc/products.d/baseproduct`. I check that it must be SLES_SAP.prod. If it could also be SLES.prod then as I understand I need to check the certificates instead?

Reply:
> @aleksei-burlakov As I have commented before, SLES is also a valid option for SAP Applications, so we should also consider vanilla SLES as a valid OS.
> 
> On the remediation text, we need something more meaningful, but we can include this later, with the support from PM.

@diegoakechi @aleksei-burlakov  SLES was intentionally not a valid option here. This check is designed to check of the OS used supports the SUSE HA Solution " SAPHanaSR Scale-up Perf-Optimized" , and that is only true for SLES4SAP.
The question about which OS is valid for SAP HANA is a different discussion and would be a different requirement. For now we are focusing on the SUSE HA context only. Hence I would not consider Vanilla SLES since this does not support SAPHanaSR.

Reply:
> As they say in https://www.suse.com/support/kb/doc/?id=000019341
> 
> _Even when the base product is SLES for SAP, the contents of the os-release file should never have been used to identify this fact._
> 
> I understand we should only look in the `/etc/products.d/baseproduct`. I check that it must be SLES_SAP.prod. If it could also be SLES.prod then as I understand I need to check the certificates instead?

@aleksei-burlakov  You need a way to distinguish between SLES and SLES4SAP; As I commented above, the intention of this check is to be sure the OS supports the HA solution SAPHanaSR  .

Reply:
> @lee-martin I think we should also link on the remediation text the SAP Note of supported OS.

@diegoakechi I'm not really keen on SUSE making statements about what SAP supports, at some point we could add a text and link pointing customers to the relevant SAP notes, but for now I'd focus on the SUSE aspects.

Reply:
@lee-martin ok, so please propose the right terminology and text for the check itself and it's remediation, because it is really hard for a user understand what is being checked here.

Reply:
> > As they say in https://www.suse.com/support/kb/doc/?id=000019341
> > _Even when the base product is SLES for SAP, the contents of the os-release file should never have been used to identify this fact._
> > I understand we should only look in the `/etc/products.d/baseproduct`. I check that it must be SLES_SAP.prod. If it could also be SLES.prod then as I understand I need to check the certificates instead?
> 
> @aleksei-burlakov You need a way to distinguish between SLES and SLES4SAP; As I commented above, the intention of this check is to be sure the OS supports the HA solution SAPHanaSR .

@lee-martin Do I understand it correctly, if the base product is the SLES_SAP.prod, it automatically supports SAPHanaSR, and no further checks needed. If it's only SLES.prog, we also need to check for support of the SAPHanaSR? If not why, then in which way I should distinguish them?

Reply:
clarified for @aleksei-burlakov that this check is specifically about the "HANA Scale-Out Performance Optimized" Scenario, so it must go in a different ruleset than the "Azure Generic" one.

Reply:
> > > As they say in https://www.suse.com/support/kb/doc/?id=000019341
> > > _Even when the base product is SLES for SAP, the contents of the os-release file should never have been used to identify this fact._
> > > I understand we should only look in the `/etc/products.d/baseproduct`. I check that it must be SLES_SAP.prod. If it could also be SLES.prod then as I understand I need to check the certificates instead?
> > 
> > 
> > @aleksei-burlakov You need a way to distinguish between SLES and SLES4SAP; As I commented above, the intention of this check is to be sure the OS supports the HA solution SAPHanaSR .
> 
> @lee-martin Do I understand it correctly, if the base product is the SLES_SAP.prod, it automatically supports SAPHanaSR, and no further checks needed. If it's only SLES.prog, we also need to check for support of the SAPHanaSR? If not why, then in which way I should distinguish them?

@lee-martin, @diegoakechi let's make it simple and assume that this yaml file is for performance-optimized scenario and check that the base-product is strictly _sles_sap.prod_ (case insensitively).

Reply:
@here It's a small PR and it breaks the next #31, let's merge it as is and we will add other checks later.

Reply:
hmm, @aleksei-burlakov this is not what we just agreed. I left you with the indication to create a separate rule set specific for the "HANA ScaleOut Perfomance Optimized" scenario. Are there other rules we already implemented that are specific to that scenario alone, and are not generic?

Reply:
> hmm, @aleksei-burlakov this is not what we just agreed. I left you with the indication to create a separate rule set specific for the "HANA ScaleOut Perfomance Optimized" scenario. Are there other rules we already implemented that are specific to that scenario alone, and are not generic?

I thought I would do it already another PR. I have amended this PR and created a new yaml file that checks only for SAPHanaSR Scale-up Perf-Optimized scenario.